libeos-payg/clock.c

```diff
@@ -0,0 +1,95 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*-
+ *
+ * Copyright © 2018 Endless Mobile, Inc.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * Alternatively, the contents of this file may be used under the terms of the
+ * GNU Lesser General Public License Version 2.1 or later (the "LGPL"), in
+ * which case the provisions of the LGPL are applicable instead of those above.
+ * If you wish to allow use of your version of this file only under the terms
+ * of the LGPL, and not to allow others to use your version of this file under
+ * the terms of the MPL, indicate your decision by deleting the provisions
+ * above and replace them with the notice and other provisions required by the
+ * LGPL. If you do not delete the provisions above, a recipient may use your
+ * version of this file under the terms of either the MPL or the LGPL.
+ */
+
+#include <clock.h>
+#include <glib.h>
+
+G_DEFINE_INTERFACE (EpgClock, epg_clock, G_TYPE_OBJECT)
+
+static void
+epg_clock_default_init (EpgClockInterface *iface)
```

Comment:
Add `gpointer iface_data` as in my other comment.

Reply:
If I do that I get
```
../libeos-payg/clock.c:26:1: error: conflicting types for ‘epg_clock_default_init’
 epg_clock_default_init (EpgClockInterface *iface,
 ^~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/gobject/gobject.h:24:0,
                 from /usr/include/glib-2.0/gobject/gbinding.h:29,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from ../libeos-payg/clock.h:22,
                 from ../libeos-payg/clock.c:20:
../libeos-payg/clock.c:23:31: note: previous declaration of ‘epg_clock_default_init’ was here
 G_DEFINE_INTERFACE (EpgClock, epg_clock, G_TYPE_OBJECT)
                               ^
/usr/include/glib-2.0/gobject/gtype.h:2008:17: note: in definition of macro ‘_G_DEFINE_INTERFACE_EXTENDED_BEGIN’
 static void     type_name##_default_init        (TypeName##Interface *klass); \
                 ^~~~~~~~~
/usr/include/glib-2.0/gobject/gtype.h:1756:47: note: in expansion of macro ‘G_DEFINE_INTERFACE_WITH_CODE’
 #define G_DEFINE_INTERFACE(TN, t_n, T_P)      G_DEFINE_INTERFACE_WITH_CODE(TN, t_n, T_P, ;)
                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
../libeos-payg/clock.c:23:1: note: in expansion of macro ‘G_DEFINE_INTERFACE’
 G_DEFINE_INTERFACE (EpgClock, epg_clock, G_TYPE_OBJECT)
 ^~~~~~~~~~~~~~~~~~
```

Reply:
Maybe the difference is that it's an interface that inherits from `GOBject` not `GAsyncInitable`

Reply:
Surely there's documentation about this somewhere but I haven't been able to find it

Reply:
Interesting. Maybe just leave out the `iface_data` arguments for now. I’ll investigate this new failure from the GLib side at some point.

> Surely there's documentation about this somewhere but I haven't been able to find it

lol no

Reply:
> > Surely there's documentation about this somewhere but I haven't been able to find it
> 
> lol no

(Finally) followed up upstream as https://gitlab.gnome.org/GNOME/glib/merge_requests/564.